### PR TITLE
Add Patience Sort

### PR DESCRIPTION
**Pull request by CascadingCascade**
#### Description of Change
Add an implementation of Patience Sort

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
[Wikipedia link to this sorting algorithm](https://en.wikipedia.org/wiki/Patience_sorting)

### CODE REVIEW

File: sorting/patience_sort.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();
+    return 0;
+}
```
Comment (Panquesito7) [inline sorting/patience_sort.c:160]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

File: sorting/patience_sort.c [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+    printf("Testing successfully completed!\n");
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();
```
Comment (Panquesito7) [inline sorting/patience_sort.c:158]:
```suggestion
    test();  // run self-test implementations
```

**Review by Panquesito7 — COMMENTED** [review]
Good work! 👍

**Review by Panquesito7 — APPROVED** [review]
Great work. Thanks! 🚀

**Review by StepfenShawn — APPROVED** [review]
Thanks for your contribution! Learned a brand new sorting algorithm for me from it.

**Comment by CascadingCascade** [comment]
> Thanks for your contribution! Learned a brand new sorting algorithm for me from it.

Glad to be of help!:smile: I should note my implementation is quite naive, it would be trivial to massively improve space usage by storing the piles with linked list / stack, but I suppose the point here is to illustrate the algorithm, if it's Java I can simply import java.util.LinkedList, but it's C

**Comment by alexpantyukhin** [comment]
Good algo! it's new for me.

**Review by alexpantyukhin — APPROVED** [review]

**Comment by CascadingCascade** [comment]
> Good algorithm implementation! it's new for me.

I suspected so!:smile: I found this algorithm awhile ago while researching some other stuff, figured it's quite cool not to mention can potentially be quite fast with enough optimizations, but then I looked around and apparently it's forgotten by everyone:thinking: This is might as well as be the first implementation on GitHub:sweat_smile: There's are implementations at [Wikibooks](https://en.wikibooks.org/wiki/Algorithm_Implementation/Sorting/Patience_sort) but they use it to find LIS instead of sort arrays

**Pull request merged**